backends-common/rabbitmq/src/test/java/org/apache/james/backend/rabbitmq/DockerRabbitMQTestRule.java

```diff
@@ -0,0 +1,44 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.backend.rabbitmq;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class DockerRabbitMQTestRule implements TestRule {
+
+    private DockerRabbitMQ dockerRabbitMQ;
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                dockerRabbitMQ = DockerRabbitMQ.withoutCookie();
+                dockerRabbitMQ.start();
+                base.evaluate();
```

Comment:
You don't stop the server?

Reply:
Note for Cleanup: We may want to implement a Cleanup task removing all the MailQueue exchanges and workQueues to achieve isolation without rebooting Rabbit nor restarting it, which takes a loooong time.

Reply:
IMO, just wait till junit 5 PRs finish. And take them. No need to spend much effort to a temporally things

Reply:
JUNIT 5 stuff can rely on a similar mechanism which would not be lost